### PR TITLE
Ignore bugfix version when running version compatibility check against Elasticsearch

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

In https://github.com/elastic/beats/pull/29683 we introduced a check to make sure we are not sending events to Elasticsearch instances that might be incompatible with Beats. However, the current check is too strict. Patch versions should not be considered during version checks. Beats 8.1.1 should be able to connect to Elasticsearch 8.1.0.

## Why is it important?

The check is too strict. This is not what we agreed on in the original issue. Furthermore, it interferes with our own release process.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

https://github.com/elastic/beats/pull/30157

### CODE REVIEW

**Review by andresrc — APPROVED** [review]

File: libbeat/common/version.go [hunk 1]
```diff
@@ -121,15 +121,28 @@ func (v *Version) LessThanOrEqual(withMeta bool, v1 *Version) bool {
...
+	if v.Minor == v1.Minor {
+		if v.Bugfix < v1.Bugfix {
+			return true
+		}
+	}
+
+	return false
+}
+
+// LessThanMajorMinor returns true if v is strictly smaller than v1. When comparing, the major,
```
Comment (ruflin) [inline libbeat/common/version.go:138]:
```suggestion
// LessThanMajorMinor returns true if v is smaller or equal to v1 based on the major and minor version. The bugfix version and meta part are not taken into account.
```

File: libbeat/cmd/instance/beat.go [hunk 2]
```diff
@@ -883,7 +883,7 @@ func (b *Beat) checkElasticsearchVersion() {
 		if err != nil {
 			return err
 		}
-		if esVersion.LessThan(beatVersion) {
+		if esVersion.LessThanMajorMinor(beatVersion) {
```
Comment (ruflin) [inline libbeat/cmd/instance/beat.go:886]: Shouldn't our comparison be LessThanOrEqual?

Comment (kvch) [reply libbeat/cmd/instance/beat.go:886]: No, we return an error if the version of ES is less than the version of Beats. If this was `LessThanOrEqual`, we would return an error when ES and Beats is on the same version.

Comment (ruflin) [reply libbeat/cmd/instance/beat.go:886]: I compared the wrong way around 🤦‍♂️

**Review by ruflin — APPROVED** [review]
LGTM

**Pull request merged**